### PR TITLE
Compose navigation in reply

### PR DESCRIPTION
**Pull request by Gurupreet**
-Add compose navigation to reply
-Add empty screens to remaining tabs

### CODE REVIEW

File: Reply/app/src/main/java/com/example/reply/ui/ReplyNavigation.kt [hunk 1]
```diff
@@ -16,9 +16,57 @@
...
 object ReplyDestinations {
     const val INBOX = "Inbox"
     const val ARTICLES = "Articles"
     const val DM = "DirectMessages"
     const val GROUPS = "Groups"
-}
\ No newline at end of file
+}
+
+class ReplyNavigationActions(navController: NavHostController) {
```
Comment (Gurupreet) [inline Reply/app/src/main/java/com/example/reply/ui/ReplyNavigation.kt:29]: I noticed JetNews used this actions implementation which seems very good for top level navigation. Hence I adopted it too. Let me know what you think on it @florina-muntenescu

File: Reply/app/src/main/java/com/example/reply/ui/EmptyComingSoon.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.reply.R
+
+@Composable
+@Preview
+fun EmptyComingSoon() {
```
Comment (Gurupreet) [inline Reply/app/src/main/java/com/example/reply/ui/EmptyComingSoon.kt:20]: There are no screen for rest of the tabs in Figma sample except Home, I have asked designer if they will implement that or it meant to only show home screen with adaptive features.

File: Reply/app/src/main/java/com/example/reply/ui/ReplyNavigation.kt (hunk 1 above)
Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyNavigation.kt:29]: https://github.com/android/compose-samples/blob/main/JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsNavigation.kt

File: Reply/app/src/main/java/com/example/reply/ui/EmptyComingSoon.kt (hunk 2 above)
Comment (ricknout) [reply Reply/app/src/main/java/com/example/reply/ui/EmptyComingSoon.kt:20]: I think it's fine for now, we've done similar in other Compose samples eg. Jetchat

File: Reply/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt [hunk 3]
```diff
@@ -50,7 +54,7 @@ class ReplyHomeViewModel(private val emailsRepository: EmailsRepository = Emails
 }
 
 data class ReplyHomeUIState(
-    val emails : List<Email> = emptyList(),
+    val emails: List<Email> = emptyList(),
```
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt:57]: Maybe you need to integrate spotless?

File: Reply/app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -11,4 +11,10 @@
     <string name="search_replies">search replies</string>
     <string name="reply">Reply</string>
     <string name="reply_all">Reply All</string>
+
+    <string name="edit">Edit</string>
+    <string name="compose">Compose</string>
+
+    <string name="empty_screen_title">Screen under construction</string>
+    <string name="empty_screen_subtitle">This screen is still under construction. This sample will help you learn about adaptive layouts in Jetpack compose</string>
```
Comment (ricknout) [inline Reply/app/src/main/res/values/strings.xml:19]: nit: "... Jetpack Compose.

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt [hunk 5]
```diff
@@ -141,54 +186,131 @@ private fun ReplyNavigationWrapperUI(
...
                 }
-            )
+            }
         }
     }
 }
 
 @Composable
-fun ReplyAppContent(
+fun RaplyNavGraph(
```
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:209]: spelling: ReplyNavGraph

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt [hunk 6]
```diff
@@ -19,16 +19,21 @@ package com.example.reply.ui
 import androidx.compose.animation.AnimatedVisibility
 import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
```
Comment (ricknout) [inline Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:22]: This file is getting quite big. Why not split out some of the composables into separate files eg. ReplyNavGraph, ReplyInboxScreen, the navigation components

File: Reply/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt (hunk 3 above)
Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt:57]: I plan to add it and also CI pipeline like other samples i think once it's merged so it's easily tested.

File: Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt (hunk 6 above)
Comment (Gurupreet) [reply Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt:22]: Divided in parts to keep navigation components and actions in separate package.

**Review by ricknout — APPROVED** [review]

**Pull request merged**